### PR TITLE
Create Grammar package

### PR DESCRIPTION
**Pull request by knothed**
Create a Grammar package just containing the `Grammar` datatype.

### CODE REVIEW

File: packages/grammar/happy-grammar.cabal [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        GHC == 8.4.4
+        GHC == 8.6.5
+        GHC == 8.8.4
+        GHC == 8.10.4
+        GHC == 9.0.1
+
+library
+  hs-source-dirs:      src
+
+  exposed-modules:     Happy.Grammar.Grammar
```
[the remark this answers is not in the document]
Comment (Ericson2314) [reply packages/grammar/happy-grammar.cabal:38]: Agreed. Sorry, meant to say so earlier.

**Comment by int-index** [comment]
LGTM overall. There are a few unnecessary formatting changes, I suggest to undo them by applying this patch

```
diff --git a/packages/grammar/src/Happy/Grammar/Grammar.lhs b/packages/grammar/src/Happy/Grammar/Grammar.lhs
index bc0822c..c30b650 100644
--- a/packages/grammar/src/Happy/Grammar/Grammar.lhs
+++ b/packages/grammar/src/Happy/Grammar/Grammar.lhs
@@ -25,7 +25,7 @@ The Grammar data type.
 
 > data Production
 >       = Production Name [Name] (String,[Int]) Priority
->       deriving (Show)
+>       deriving Show
 
 > data Grammar
 >       = Grammar {
@@ -123,4 +123,4 @@ In hindsight, this was probably a bad idea.
 > firstStartTok   = 3
 > dummyTok        = 2
 > errorTok        = 1
-> epsilonTok      = 0
\ No newline at end of file
+> epsilonTok      = 0
diff --git a/src/LALR.lhs b/src/LALR.lhs
index b1d746e..a6e5473 100644
--- a/src/LALR.lhs
+++ b/src/LALR.lhs
@@ -6,7 +6,7 @@ Generation of LALR parsing tables.
 -----------------------------------------------------------------------------
 
 > module LALR
->       (ActionTable, GotoTable, 
+>       (ActionTable, GotoTable,
 >        genActionTable, genGotoTable, genLR0items, precalcClosure0,
 >        propLookaheads, calcLookaheads, mergeLookaheadInfo, countConflicts,
 >        Lr0Item(..), Lr1Item(..), ItemSetWithGotos, LRAction(..), Goto(..))
@@ -53,16 +53,12 @@ This means rule $a$, with dot at $b$ (all starting at 0)
 >               | LR'Fail               -- :-(
 >               | LR'MustFail           -- :-(
 >               | LR'Multiple [LRAction] LRAction       -- conflict
->       deriving (Eq, Show)
+>       deriving (Eq,Show)
 
-> type ActionTable
->      = Array Int{-state-} (Array Int{-terminal#-} LRAction)
-> type GotoTable
->      = Array Int{-state-} (Array Int{-nonterminal #-} Goto)
-> data Goto
->      = Goto Int
->      | NoGoto
->      deriving(Eq, Show)
+> type ActionTable = Array Int{-state-} (Array Int{-terminal#-} LRAction)
+> type GotoTable = Array Int{-state-} (Array Int{-nonterminal #-} Goto)
+> data Goto = Goto Int | NoGoto
+>       deriving (Eq, Show)
 
 -----------------------------------------------------------------------------
 Generating the closure of a set of LR(0) items
diff --git a/src/ProduceCode.lhs b/src/ProduceCode.lhs
index 4a547f1..b5e9cfd 100644
--- a/src/ProduceCode.lhs
+++ b/src/ProduceCode.lhs
@@ -10,7 +10,7 @@ The code generator.
 > import Data.Version           ( showVersion )
 > import Happy.Grammar.Grammar
 > import Target                 ( Target(..) )
-> import LALR
+> import LALR                   ()
 > import GenUtils               ( mapDollarDollar, str, char, nl, strspace,
 >                                 interleave, interleave', maybestr,
 >                                 brack, brack' )
```

This way `git diff master --color-moved=blocks -w` can detect more changes as moved blocks rather than removed/added code.

**Comment by Ericson2314** [comment]
We can always do the inverse of that patch as follow up commit/PR :)

**Comment by knothed** [comment]
Okay done. :)

File: src/Main.lhs [hunk 2]
```diff
@@ -1,5 +1,4 @@
------------------------------------------------------------------------------
```
Comment (int-index) [inline src/Main.lhs:1]: accidental removal?

**Comment by int-index** [comment]
Thanks!

**Pull request merged**